### ISSUE TITLE
"use" directive in language.xml in language package does not work properly.

### ISSUE THREAD

**Issue by LordHansolo** [magento/magento2#20150]
### Preconditions (*)
1. Magento 2.2.5 CE

### Steps to reproduce (*)
1. Install 2 language package for example. [Snowdog Polish language pack](https://github.com/SnowdogApps/magento2-pl_pl) and [Mageplaza language pack](https://github.com/mageplaza/magento-2-polish-language-pack)
2. Create custom language package in app/i18n directory.
3. Add "sort_order" directive with value like 120.
4. Add "use" directive with Snowdog and Mageplaza language packages inside language.xml.

`<?xml version="1.0"?>
<language xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:App/Language/package.xsd">
    <code>pl_PL</code>
    <vendor>custom</vendor>
    <package>pl_pl</package>
    <sort_order>120</sort_order>
    <use vendor="snowdog" package="pl_pl"/>
    <use vendor="mageplaza" package="pl_pl"/>
</language>
`

### Expected result (*)
1. Custom language package is loaded after Snowdog and Mageplaza language package.
2. Magento first looks phrases translations inside custom language package
3. If specified phrase is not found Magento looks for that phrase inside parent language packages in order specified in language.xml. In our case Magento should looks phrase translation in order "custom language package"->"snowdog language package"->"mageplaza language package"

### Actual result (*)
1. Custom language package is loaded after Snowdog and Mageplaza language package.
2. Magento first looks phrases translations inside custom language package.
3. If phrase is not found Magento looks for that phrase in parent language packages but in random order without attention on parent packages order.

### Additional info
Language package load order is described in https://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-i18n.html

**Comment by magento-engcom-team** [comment]
Hi @LordHansolo. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@LordHansolo do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by hostep** [comment]
This might be a duplicate of https://github.com/magento/magento2/issues/8691

**Comment by milindsingh** [comment]
@hostep Yes, it's duplicate. I am closing this one.
